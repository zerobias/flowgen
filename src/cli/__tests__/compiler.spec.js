/* @flow */
import compiler from "../compiler";

it("should handle maybe & nullable type", () => {
  const result = compiler.compileDefinitionString(
    "let a: string | null | undefined",
  );

  expect(result).toMatchSnapshot();
});

describe("should handle true and false types", () => {
  it('in union', () => {
    const result = compiler.compileDefinitionString(
      "let a: true | false;",
    );

    expect(result).toMatchSnapshot();
  })

  describe('in single type', () => {
    test('true', () => {
      const result = compiler.compileDefinitionString(
        "let a: true;",
      );

      expect(result).toMatchSnapshot();
    })

    test('true', () => {
      const result = compiler.compileDefinitionString(
        "let a: false;",
      );

      expect(result).toMatchSnapshot();
    })
  })
})

it("should handle bounded polymorphism", () => {
  const ts = `
    function fooGood<T extends { x: number }>(obj: T): T {
      console.log(Math.abs(obj.x));
      return obj;
    }
  `;

  const result = compiler.compileDefinitionString(ts);

  expect(result).toMatchSnapshot();
});
