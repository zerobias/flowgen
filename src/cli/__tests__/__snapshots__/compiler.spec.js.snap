// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle bounded polymorphism 1`] = `
"declare function fooGood<T>(obj: T): T
"
`;

exports[`should handle maybe & nullable type 1`] = `"declare var a: string | null | void;"`;

exports[`should handle true and false types in single type true 1`] = `"declare var a: true;"`;

exports[`should handle true and false types in single type true 2`] = `"declare var a: false;"`;

exports[`should handle true and false types in union 1`] = `"declare var a: true | false;"`;
